### PR TITLE
cctools: add 7.4.2

### PR DESCRIPTION
**Pull request by dthain**
- Update cctools to 7.4.2
- Patch two test cases.

### CODE REVIEW

File: var/spack/repos/builtin/packages/cctools/package.py [hunk 1]
```diff
@@ -13,8 +15,10 @@ class Cctools(AutotoolsPackage):
     """
 
     homepage = "https://cctools.readthedocs.io"
-    url      = "https://ccl.cse.nd.edu/software/files/cctools-7.2.4-source.tar.gz"
+    url      = "https://ccl.cse.nd.edu/software/files/cctools-7.4.2-source.tar.gz"
+    maintainers = ['dthain', 'btovar']
 
+    version('7.4.2', sha256='7c8c86f09bcb9ad23ab44e52dfb5bf3a4e4a1ad84ef51e500a9c8c2371770f4b')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/cctools/package.py:21]: Confirmed sha256

File: var/spack/repos/builtin/packages/cctools/package.py [hunk 2]
```diff
@@ -56,6 +60,16 @@ def patch(self):
...
 
+        # Remove two troublesome test cases:
+
+        # TR_parrot_dir.sh fails b/c of variations in how Linux kernels
+        # deal with open() on directories.
+        os.unlink("parrot/test/TR_parrot_dir.sh")
+
+        # TR_parrot_execve.sh fails b/c of the complexities of moving
+        # a python interpreter around.
+        os.unlink("parrot/test/TR_parrot_execve.sh")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/cctools/package.py:71]: Are these tools present in every version of the package?  If they may not exist in some version, then you'll want to check first.

**Review by tldahlgren — CHANGES_REQUESTED** [review]

**Comment by tldahlgren** [comment]
@spackbot run pipeline

File: var/spack/repos/builtin/packages/cctools/package.py (hunk 2 above)
Comment (dthain) [reply var/spack/repos/builtin/packages/cctools/package.py:71]: Yes, these are integral test scripts in the repo.  it turns out that they are rather platform sensitive, so I'm removing them here while we ponder how to fix them upstream..

**Comment by dthain** [comment]
Hello @tldahlgren -- are you waiting on something from me here? 
Sorry to be a pest, I haven't done one of these before.

**Review by tldahlgren — APPROVED** [review]

**Comment by tldahlgren** [comment]
> Hello @tldahlgren -- are you waiting on something from me here?
> Sorry to be a pest, I haven't done one of these before.

Np.  Apologies for losing track.  Feel free to ping me again if I don't respond in a week or two.

**Pull request merged**